core/chains/evm/txmgr/stuck_tx_detector_test.go

```diff
@@ -280,6 +280,95 @@ func TestStuckTxDetector_DetectStuckTransactionsHeuristic(t *testing.T) {
+		attempts := tx.TxAttempts[0]
+		// Request still returns transaction by hash, transaction not discarded by network and not considered stuck
+		ethClient.On("BatchCallContext", mock.Anything, mock.MatchedBy(func(b []rpc.BatchElem) bool {
+			return len(b) == 1 && cltest.BatchElemMatchesParams(b[0], attempts.Hash, "zirc_isQuarantined")
+		})).Return(nil).Run(func(args mock.Arguments) {
+			elems := args.Get(1).([]rpc.BatchElem)
+			resp, err := json.Marshal(struct {
+				IsQuarantined bool `json:"isQuarantined"`
+			}{IsQuarantined: false})
+			require.NoError(t, err)
+			elems[0].Error = json.Unmarshal(resp, elems[0].Result)
+		}).Once()
+
+		txs, err := stuckTxDetector.DetectStuckTransactions(ctx, []common.Address{fromAddress}, blockNum)
+		require.NoError(t, err)
+		require.Len(t, txs, 0)
+	})
+
+	t.Run("returns fraud transactions identified", func(t *testing.T) {
+		_, fromAddress := cltest.MustInsertRandomKey(t, ethKeyStore)
+		tx := mustInsertUnconfirmedTxWithBroadcastAttempts(t, txStore, 0, fromAddress, 1, blockNum, tenGwei)
+		attempts := tx.TxAttempts[0]
+		// Request still returns transaction by hash, transaction not discarded by network and not considered stuck
+		ethClient.On("BatchCallContext", mock.Anything, mock.MatchedBy(func(b []rpc.BatchElem) bool {
+			return len(b) == 1 && cltest.BatchElemMatchesParams(b[0], attempts.Hash, "zirc_isQuarantined")
+		})).Return(nil).Run(func(args mock.Arguments) {
+			elems := args.Get(1).([]rpc.BatchElem)
+			resp, err := json.Marshal(struct {
+				IsQuarantined bool `json:"isQuarantined"`
+			}{IsQuarantined: true})
+			require.NoError(t, err)
+			elems[0].Error = json.Unmarshal(resp, elems[0].Result)
+		}).Once()
+
+		txs, err := stuckTxDetector.DetectStuckTransactions(ctx, []common.Address{fromAddress}, blockNum)
+		require.NoError(t, err)
+		require.Len(t, txs, 1)
+	})
+
+	t.Run("returns the transaction only once if it's identified as both fraud and stuck", func(t *testing.T) {
+		_, fromAddress := cltest.MustInsertRandomKey(t, ethKeyStore)
+		tx := mustInsertUnconfirmedTxWithBroadcastAttempts(t, txStore, 0, fromAddress, autoPurgeMinAttempts, blockNum-int64(autoPurgeThreshold)+int64(autoPurgeMinAttempts-1), marketGasPrice.Add(oneGwei))
+		attempts := tx.TxAttempts[0]
+
+		ethClient.On("BatchCallContext", mock.Anything, mock.MatchedBy(func(b []rpc.BatchElem) bool {
+			return len(b) == 1 && cltest.BatchElemMatchesParams(b[0], attempts.Hash, "zirc_isQuarantined")
+		})).Return(nil).Run(func(args mock.Arguments) {
+			elems := args.Get(1).([]rpc.BatchElem)
+			resp, err := json.Marshal(struct {
+				IsQuarantined bool `json:"isQuarantined"`
+			}{IsQuarantined: true})
+			require.NoError(t, err)
+			elems[0].Error = json.Unmarshal(resp, elems[0].Result)
+		}).Once()
+
+		txs, err := stuckTxDetector.DetectStuckTransactions(ctx, []common.Address{fromAddress}, blockNum)
+		require.NoError(t, err)
+		require.Len(t, txs, 1)
+	})
```

Comment:
Once you address my previous comment, it might be worth adding a test to check if we still continue to overflow detection even if we get an error during fraud detection